build-binutils-2.30.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+# NOTE: This script ONLY builds a Linux 64-bit binary! OSX uses the XCode
+# command-line tools installed on the system to link executables.
+
+if ! hash xz; then
+  cat >&2 <<EOF
+'xz' is required to run this script. You may have to install it using your
+operating system's package manager.
+EOF
+  exit 1
+fi
+
+set -euxo pipefail
+
+BINUTILS_SUPPORTDIR='build-support/bin/binutils'
+BINUTILS_PANTS_ARCHIVE_NAME='binutils.tar.gz'
+BINUTILS_BUILD_TMP_DIR='binutils-tmp'
+BINUTILS_INSTALL_DIRNAME='binutils-install'
+
+BINUTILS_SRC_DIRNAME="binutils-${BINUTILS_VERSION}"
+BINUTILS_RELEASE_ARCHIVE_NAME="binutils-${BINUTILS_VERSION}.tar.xz"
+BINUTILS_RELEASE_URL="https://ftpmirror.gnu.org/binutils/${BINUTILS_RELEASE_ARCHIVE_NAME}"
+
+# default to -j2
+MAKE_JOBS="${MAKE_JOBS:-2}"
+
+## Linux (binutils, from source release)
+BINUTILS_VERSION='2.30'
+BINUTILS_TMP_ARCHIVE_CREATION_DIR='binutils-tmp'
+
+mkdir -p "$BINUTILS_BUILD_TMP_DIR"
+pushd "$BINUTILS_BUILD_TMP_DIR" # root -> $BINUTILS_BUILD_TMP_DIR
+
+curl -L -v -O "$BINUTILS_RELEASE_URL"
```

Comment:
Throw in a `--fail`, and throughout the PR (which I really, really wish was the default)

Reply:
[Done](https://github.com/cosmicexplorer/binaries/blob/10e4d82f885b9c8cf2311239be898d6bec548426/utils.bash#L64)!